### PR TITLE
First package version (with icon sprite generation)

### PR DESCRIPTION
**Pull request by jouderianjr**
Paack-ui needs some non-Elm assets and because of that, we have to create a project to hold those assets.

We have been using this as a test from my personal account in npm, now that I got the npm paackEng credentials, I'm trying to make the things correctly 💩 .

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "paack-ui-assets",
+  "version": "0.0.11",
+  "description": "",
+  "files": ["js"],
+  "scripts": {
+    "prepublish": "make build"
+  },
+  "author": "PackEng",
+  "license": "ISC",
```
Comment (miguelcoba) [inline package.json:10]: is this the correct license?

File: src/js/paackSvgIconSprite.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { readFileSync } from 'fs';
+
+const svg = readFileSync('./dist/icon-sprite.svg', 'utf-8');
+
+class PaackSvgIconSprite extends HTMLElement {
+  constructor() {
```
Comment (miguelcoba) [inline src/js/paackSvgIconSprite.js:6]: I think the constructor that only calls super can be omitted

**Review by miguelcoba — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+	npx parcel build src/js/paackSvgIconSprite.js --out-dir ./js
+
+
+build-svg-icon-sprite: optimize-svgs create-svg-sprite
+
+optimize-svgs:
+	npx svgo --config=./svgo.json --folder=./src/svg --output=./dist/optimizedSvg --multipass
+
+create-svg-sprite:
+	npx svg-sprite --symbol --symbol-inline --symbol-dest=dist/ --symbol-sprite=icon-sprite.svg dist/optimizedSvg/*.svg
```
Comment (renanpvaz) [inline Makefile:14]: we could just install these packages and use them inside npm scripts. NPM looks for the binaries inside `node_modules` so it should be a little faster than using `npx` and we keep all the scripts on the same place

File: package.json (hunk 1 above)
Comment (jouderianjr) [reply package.json]: @miguelcoba I've changed to MIT and added the license file.

File: src/js/paackSvgIconSprite.js (hunk 2 above)
Comment (jouderianjr) [reply src/js/paackSvgIconSprite.js]: It's true, nice catch

File: Makefile (hunk 3 above)
Comment (jouderianjr) [reply Makefile:14]: Do you mean to move to do something like this?

```makefile
optimize-svgs:
    npm optimize-svgs

create-svg-sprite:
    npm create-svg-sprite
```

Comment (renanpvaz) [reply Makefile:14]: inside `package.json` having `parcel`, `svgo` and `svg-sprite` as dependencies

```json
"scripts": {
  "build": "parcel build...",
  "optimize-svgs": "svgo --config...",
  "create-svg-sprite": "svg-sprite --symbol..." 
}

**Review by miguelcoba — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+build-svg-icon-sprite: optimize-svgs create-svg-sprite
+
+optimize-svgs:
+	npm run optimize-svgs
+
+create-svg-sprite:
+	npm run create-svg-sprite
+
+.PHONY: build-svg-icon-sprite publish optimize-svgs create-svg-sprite
```
Comment (PedroHLC) [inline Makefile:16]: (if not removing this file)
add `build` too, whenever a rule doesn't create with the same name, it must be on `.PHONY` :smile:

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,16 @@
+publish:
+	npm publish
+
+build: build-svg-icon-sprite
+	npm run build
+
+
+build-svg-icon-sprite: optimize-svgs create-svg-sprite
```
Comment (PedroHLC) [inline Makefile:8]: We only have one rule not covered by package.json
I would move it to there under `npm run build-spritesheet`, and then delete this file.

**Pull request merged**